### PR TITLE
ADEN-4180 Override community skins for Fan Takeover

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [bfaaDesktop.js#L13](https://github.com/Wikia/app/blob/4686b4a65efd730f8c91949c1b91452c426380a5/extensions/wikia/AdEngine/js/template/bfaaDesktop.js#L13): Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [bfaaDesktop.js#L13](https://github.com/Wikia/app/blob/209f91a3a65d5759c15a7f1676150682011af66f/extensions/wikia/AdEngine/js/template/bfaaDesktop.js#L13): Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: skins/shared/functions.scss [hunk 1]
```diff
@@ -11,3 +11,94 @@
 	}
 	@return nth($range, 2);
 }
+
+@function luminance($color) {
```
Comment (fraszczakszymon) [inline skins/shared/functions.scss:15]: Alternatives:
1. Download it as a dependency, but we don't have clear way to do it.
2. Move implementation to JS, but then luminance will be calculated each time on client side.
3. Use built-in sass lightness method, which has a different meaning. It won't meet spec fully, but I think it fits to this case.
4. Simplify luminance calculation to: `0.3*Red + 0.59*Green + 0.11*Blue`, which is not really true :ugly:.

File: skins/oasis/css/core/breakpoints-background.scss [hunk 2]
```diff
@@ -178,3 +178,22 @@ body.skin-oasis {
 		@include background-max;
 	}
 }
+
+body.skin-oasis.uap-skin {
+	$uap-color: mix(white, $color-body, 15%);
```
Comment (nandy-andy) [inline skins/oasis/css/core/breakpoints-background.scss:183]: ~~[Mr. White](https://github.com/Wikia/app/blob/dev/extensions/wikia/DesignSystem/bower_components/design-system/dist/scss/wds-variables/_colors.scss#L42)~~ Ignore. Discussed on Slack.

File: skins/oasis/css/core/breakpoints-background.scss [hunk 3]
```diff
@@ -178,3 +178,22 @@ body.skin-oasis {
...
+	@if (luminance($color-body) > 0.3) {
+		$uap-color: mix(black, $color-body, 10%);
```
Comment (nandy-andy) [inline skins/oasis/css/core/breakpoints-background.scss:185]: ~~And we can try with [Mr. Black](https://github.com/Wikia/app/blob/dev/extensions/wikia/DesignSystem/bower_components/design-system/dist/scss/wds-variables/_colors.scss#L33)~~ Ignore. Discussed on Slack.

File: skins/shared/functions.scss (hunk 1 above)
Comment (nandy-andy) [reply skins/shared/functions.scss:15]: Thanks for providing them. How are you plan to pick an alternative if decided to pick one? ;) Also, a suggestion to write unit tests if we decide to stay with SASS implementation below ;)

**Review by nandy-andy — APPROVED** [review]

**Pull request merged**